### ISSUE TITLE
Cannot Build Android after upgrading to 1.0.0

### ISSUE THREAD

**Issue by joncursi** [zoontek/react-native-permissions#117]
After upgrading to `1.0.0`, I get the following output when creating a release build of a react native android app:

```
22:45:12.675 [DEBUG] [org.gradle.model.internal.registry.DefaultModelRegistry] Project :react-native-permissions - Registering model element 'tasks.help' (hidden = false)
22:45:12.675 [DEBUG] [org.gradle.model.internal.registry.DefaultModelRegistry] Project :react-native-permissions - Registering model element 'tasks.projects' (hidden = false)
22:45:12.675 [DEBUG] [org.gradle.model.internal.registry.DefaultModelRegistry] Project :react-native-permissions - Registering model element 'tasks.tasks' (hidden = false)
22:45:12.675 [DEBUG] [org.gradle.model.internal.registry.DefaultModelRegistry] Project :react-native-permissions - Registering model element 'tasks.properties' (hidden = false)
22:45:12.675 [DEBUG] [org.gradle.model.internal.registry.DefaultModelRegistry] Project :react-native-permissions - Registering model element 'tasks.dependencyInsight' (hidden = false)
22:45:12.675 [DEBUG] [org.gradle.model.internal.registry.DefaultModelRegistry] Project :react-native-permissions - Registering model element 'tasks.dependencies' (hidden = false)
22:45:12.675 [DEBUG] [org.gradle.model.internal.registry.DefaultModelRegistry] Project :react-native-permissions - Registering model element 'tasks.buildEnvironment' (hidden = false)
22:45:12.675 [DEBUG] [org.gradle.model.internal.registry.DefaultModelRegistry] Project :react-native-permissions - Registering model element 'tasks.components' (hidden = false)
22:45:12.675 [DEBUG] [org.gradle.model.internal.registry.DefaultModelRegistry] Project :react-native-permissions - Registering model element 'tasks.model' (hidden = false)
22:45:12.675 [INFO] [org.gradle.configuration.project.BuildScriptProcessor] Evaluating project ':react-native-permissions' using build file '/Users/jcursi/Sites/joncursi/redbirdNative/node_modules/react-native-permissions/android/build.gradle'.
22:45:12.676 [DEBUG] [org.gradle.configuration.project.BuildScriptProcessor] Timing: Running the build script took 0.001 secs
22:45:12.676 [DEBUG] [org.gradle.model.internal.registry.DefaultModelRegistry] Project :react-native-permissions - Transitioning model element 'tasks' from state Created to SelfClosed
22:45:12.676 [DEBUG] [org.gradle.model.internal.registry.DefaultModelRegistry] Project :react-native-permissions - Transitioning model element 'tasks' to state DefaultsApplied.
22:45:12.676 [DEBUG] [org.gradle.model.internal.registry.DefaultModelRegistry] Project :react-native-permissions - Transitioning model element 'tasks' to state Initialized.
22:45:12.676 [DEBUG] [org.gradle.model.internal.registry.DefaultModelRegistry] Project :react-native-permissions - Transitioning model element 'tasks' to state Mutated.
22:45:12.676 [DEBUG] [org.gradle.model.internal.registry.DefaultModelRegistry] Project :react-native-permissions - Transitioning model element 'tasks' to state Finalized.
22:45:12.676 [DEBUG] [org.gradle.model.internal.registry.DefaultModelRegistry] Project :react-native-permissions - Transitioning model element 'tasks' to state SelfClosed.
22:45:12.686 [ERROR] [org.gradle.BuildExceptionReporter] 
22:45:12.687 [ERROR] [org.gradle.BuildExceptionReporter] FAILURE: Build failed with an exception.
22:45:12.687 [ERROR] [org.gradle.BuildExceptionReporter] 
22:45:12.687 [ERROR] [org.gradle.BuildExceptionReporter] * What went wrong:
22:45:12.687 [ERROR] [org.gradle.BuildExceptionReporter] A problem occurred configuring project ':app'.
22:45:12.687 [ERROR] [org.gradle.BuildExceptionReporter] > Could not resolve all dependencies for configuration ':app:_debugApk'.
22:45:12.687 [ERROR] [org.gradle.BuildExceptionReporter]    > Configuration with name 'default' not found.
22:45:12.687 [ERROR] [org.gradle.BuildExceptionReporter] 
22:45:12.687 [ERROR] [org.gradle.BuildExceptionReporter] * Try:
22:45:12.687 [ERROR] [org.gradle.BuildExceptionReporter] Run with --stacktrace option to get the stack trace. 
22:45:12.687 [LIFECYCLE] [org.gradle.BuildResultLogger] 
22:45:12.687 [LIFECYCLE] [org.gradle.BuildResultLogger] BUILD FAILED
22:45:12.687 [LIFECYCLE] [org.gradle.BuildResultLogger] 
22:45:12.687 [LIFECYCLE] [org.gradle.BuildResultLogger] Total time: 11.605 secs
22:45:12.838 [DEBUG] [org.gradle.api.internal.artifacts.ivyservice.ivyresolve.memcache.InMemoryCachedRepositoryFactory] In-memory dependency metadata cache closed. Repos cached: 1129, cache instances: 9, modules served from cache: 16851, artifacts: 4041
22:45:12.838 [DEBUG] [org.gradle.cache.internal.DefaultCacheAccess] Cache Generated Gradle JARs cache (/Users/jcursi/.gradle/caches/2.14.1/generated-gradle-jars) was closed 0 times.
22:45:12.838 [DEBUG] [org.gradle.cache.internal.btree.BTreePersistentIndexedCache] Closing cache module-metadata.bin (/Users/jcursi/.gradle/caches/modules-2/metadata-2.16/module-metadata.bin)
22:45:12.839 [DEBUG] [org.gradle.cache.internal.btree.BTreePersistentIndexedCache] Closing cache module-artifacts.bin (/Users/jcursi/.gradle/caches/modules-2/metadata-2.16/module-artifacts.bin)
22:45:12.839 [DEBUG] [org.gradle.cache.internal.btree.BTreePersistentIndexedCache] Closing cache artifact-at-repository.bin (/Users/jcursi/.gradle/caches/modules-2/metadata-2.16/artifact-at-repository.bin)
22:45:12.839 [DEBUG] [org.gradle.cache.internal.btree.BTreePersistentIndexedCache] Closing cache module-versions.bin (/Users/jcursi/.gradle/caches/modules-2/metadata-2.16/module-versions.bin)
22:45:12.839 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Releasing lock on artifact cache (/Users/jcursi/.gradle/caches/modules-2).
22:45:12.907 [DEBUG] [org.gradle.api.internal.artifacts.ivyservice.resolveengine.store.CachedStoreFactory] Resolution result cache closed. Cache reads: 0, disk reads: 0 (avg: 0.0 secs, total: 0.0 secs)
22:45:12.909 [DEBUG] [org.gradle.api.internal.artifacts.ivyservice.resolveengine.store.CachedStoreFactory] Resolution result cache closed. Cache reads: 0, disk reads: 160 (avg: 0.0 secs, total: 0.138 secs)
22:45:12.910 [DEBUG] [org.gradle.api.internal.artifacts.ivyservice.resolveengine.store.ResolutionResultsStoreFactory] Deleted 2 resolution results binary files in 0.07 secs
22:45:12.910 [DEBUG] [org.gradle.cache.internal.DefaultCacheAccess] Cache Plugin Resolution Cache (/Users/jcursi/.gradle/caches/2.14.1/plugin-resolution) was closed 0 times.
22:45:12.911 [DEBUG] [org.gradle.api.internal.tasks.compile.daemon.CompilerDaemonManager] Stopping 0 compiler daemon(s).
22:45:12.911 [INFO] [org.gradle.api.internal.tasks.compile.daemon.CompilerDaemonManager] Stopped 0 compiler daemon(s).
```

**Comment by yonahforst** [comment]
@joncursi you may need to unlink and then link again. Try `react-native unlink react-native-permissions` then `react-native link react-native-permissions`

**Comment by joncursi** [comment]
Hmm... no luck. When I do that, `react-native link` only seems to take action on iOS, nothing about linking / unlinking Android gets logged out:

```
Jon-Cursis-ProExamMac:redbirdNative jcursi$ react-native unlink react-native-permissions
Scanning 1083 folders for symlinks in /Users/jcursi/Sites/joncursi/redbirdNative/node_modules (7ms)
rnpm-install info Unlinking react-native-permissions ios dependency 
rnpm-install info iOS module react-native-permissions has been successfully unlinked 
Jon-Cursis-ProExamMac:redbirdNative jcursi$ react-native link react-native-permissions
Scanning 1083 folders for symlinks in /Users/jcursi/Sites/joncursi/redbirdNative/node_modules (6ms)
rnpm-install info Linking react-native-permissions ios dependency 
rnpm-install info iOS module react-native-permissions has been successfully linked 
```

Any ideas on why that is?

**Comment by yonahforst** [comment]
Hrmm. Maybe because I've already removed android references from the project, it thinks there's nothing to unlink. 

Probably your best bet would be to open android studio and just remove references to react native permissions. Gradle should guide you by throwing errors at every reference you need to remove.

**Comment by yonahforst** [comment]
@joncursi  - any luck with this?

**Comment by jakegardner** [comment]
i can confirm doing the manual unlinking allows it to build.

settings.gradle
```
-include ':react-native-permissions'
-project(':react-native-permissions').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-permissions/android')
```
MainApplication.java
```
-import com.joshblour.reactnativepermissions.ReactNativePermissionsPackage;
```
```
-            new ReactNativePermissionsPackage(),
```
app/build.gradle
```
-    compile project(':react-native-permissions')
```

**Comment by joncursi** [comment]
I was able to get the project to build by following the above steps mentioned by @jakegardner , with the only exception of having to additionally remove:

MainActivity.java
```
import com.joshblour.reactnativepermissions.ReactNativePermissionsPackage;
```

So now that react-native-permissions was completely removed from Android code, I tried `react-native link react-native-permissions` to try and re-setup Android support, but it's still ignoring Android setup altogether:

```
$ react-native link react-native-permissions
Scanning 1084 folders for symlinks in /Users/jcursi/Sites/joncursi/redbirdNative/node_modules (7ms)
rnpm-install info iOS module react-native-permissions is already linked 
```

No references get added back in to the `android` directory.

I tried opening up Android Studio and poking around there, but the interface is pretty confusing and I wasn't sure what to look for.

**Comment by joncursi** [comment]
Ohhhhhh, wait a minute. I just realized that https://github.com/yonahforst/react-native-permissions/commit/306e6928c1dba18ff46c9ff2daf09f2c4e015bad deliberately wants Android references to be removed. I thought we still needed them, but `react-native link` was not cooperating with us. I was misunderstanding the problem / exactly what was breaking in the new release.

So in this case, after manually removing the above references, I think the problem is now solved for me since gradle successfully builds again! Will release a new test app in the next few days and verify on a physical device.
